### PR TITLE
Add custom converter feature

### PR DESCRIPTION
**Pull request by Jamesgt**

### CODE REVIEW

File: src/date.ts [hunk 1]
```diff
@@ -8,15 +8,14 @@ interface CoerceOptions {
 interface ValidationOptions extends CommonValidationOptions<Date> {
   maxFuture?: Duration;
   maxPast?: Duration;
+  format?: string;
```
Comment (gregbacchus) [inline src/date.ts]: Is this supposed to be here? Seems not be be used, but would be more at home in the convertOptions.

File: src/common.ts [hunk 2]
```diff
@@ -60,8 +60,13 @@ export const is = <T>(check: Check<T>, typeName: string): IsAs<T> => (next) => (
   return next(value);
 };
 
-export const as = <T>(convert: Convert<T>, typeName: string, undefinedHandler?: UndefinedHandler<T>): IsAs<T> => (next) => (value) => {
-  const converted = convert(value);
+export const as = <T, O extends CommonConvertOptions<T>>(check: Check<T>, convert: Convert<T, O>, typeName: string, undefinedHandler?: UndefinedHandler<T>, options?: O): IsAs<T> => (next) => (value) => {
+  if (check(value)) return next(value);
+
+  // const converted = options?.converter
```
Comment (gregbacchus) [inline src/common.ts]: commented code

Comment (Jamesgt) [reply src/common.ts]: removed

File: src/date.ts (hunk 1 above)
Comment (Jamesgt) [reply src/date.ts]: no and you gave the idea to include it as a built-in convert option, thx

**Pull request merged**